include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -167,43 +168,50 @@ class pairwise_alignment_algorithm : protected policies_t...
 
         this->reset_optimum(); // Reset the tracker for the new alignment computation.
 
-        thread_local score_matrix_single_column<int32_t> local_matrix{};
-        local_matrix.reset_matrix(std::forward<sequence1_t>(sequence1), std::forward<sequence2_t>(sequence2));
-        auto matrix_iter = local_matrix.begin();
+        thread_local score_matrix_single_column<int32_t> local_score_matrix{};
+        thread_local coordinate_matrix local_coordinate_matrix{};
 
-        initialise_column(*matrix_iter, sequence2);
+        local_score_matrix.reset_matrix(sequence1, sequence2);
```

Comment:
> Looking at the interface, are the sequences actually really stored in there? If not, would it not suffice to change the interface to resize(size_t rows, size_t columns)? And have that more similar to a vector?

As I mentioned before. Not needed yet. And I wanted to remove this as well. Just wasn't sure if I might need it later. But also because I need to call distance only once, especially if it is a non-sized and non-sized_sentinel range